### PR TITLE
Add type flags support for `Ty` and `Const` late-bound variables

### PR DESCRIPTION
**Pull request by compiler-errors**
I've been working on `for<T>` binders, and these will eventually be useful.

### CODE REVIEW

File: compiler/rustc_middle/src/ty/flags.rs [hunk 1]
```diff
@@ -60,7 +60,19 @@ impl FlagComputation {
         let mut computation = FlagComputation::new();
 
         if !value.bound_vars().is_empty() {
```
Comment (fbstj) [inline compiler/rustc_middle/src/ty/flags.rs:62]: no longer needs this check for the inner `for` loop?

**Comment by cjgillot** [comment]
Could you elaborate the use case for these `for<T>` binders? Predicates only, or do you have plans to have them in types too?

**Comment by compiler-errors** [comment]
@cjgillot yeah, predicates are probably the only place they make sense in the language currently.

**Comment by cjgillot** [comment]
r=me with @fbstj's suggestion applied.

**Comment by compiler-errors** [comment]
@bors retry

That test timeout looks spurious?

**Pull request merged**